.github/workflows/ci_pipeline.yml

```diff
@@ -46,9 +46,15 @@ jobs:
         poetry run vulndb nvd ingest
         poetry run vulndb osv ingest
         poetry run poe data_validation_test
-    - name: Store database artifcats
-      if: ${{ github.event_name == 'workflow_dispatch' }}
-      uses: actions/upload-artifact@v4
+    - name: Save Database to Cache
+      id: cache-database-if-valid
+      uses: actions/cache/save@v4
+      with:
+        key: ${{ github.head_ref }}.${{ github.sha }.db}
+        path: |
+          ~/.vulndb/vuln.db
+    - name: Upload asset to Release
+      uses: softprops/action-gh-release@v2
+      if: github.ref == 'refs/heads/master'
       with:
-        name: database
-        path: ~/.vulndb/vuln.db
```

Comment:
How large does this end up being? We likely want to compress it. How does it end up being versioned as a release asset?

Reply:
All cache uses zstd compression by default. It is not currently versioned beyond the branch_hash naming convention. Updates to the database are not currently pushed to the release asset either. Those are a work in progress. This is to get a baseline sense of creating and using the database as a release asset and running the update as a scheduled github action.

Reply:
what about compressing the release asset? Without compression, it should be around 2.6 GB, after compression around 300 MB

Reply:
^^ that - we should store a compressed artifact so that one does not have to download a 2.5Gb+ artifact

Reply:
it is compressed